### PR TITLE
Adding replicator and reconciling changes with the latest c-api, reso…

### PR DESCRIPTION
**Pull request by bhtaulbee**
…lving multiple breakages.

### CODE REVIEW

File: CouchbaseLite/Document.py [hunk 1]
```diff
@@ -78,7 +78,9 @@ def getProperties(self):
 
     @property
     def JSON(self):
-        return sliceToString(lib.CBLDocument_PropertiesAsJSON(self._ref))
+        # TODO: c-api call doesn't exist any more, CBLDocument_PropertiesAsJSON
+        #       need to test the following to see if it works
+        return sliceToString(lib.CBLDocument_Properties(self._ref))
```
Comment (snej) [inline CouchbaseLite/Document.py]: That will not work. CBLDocument_Properties returns a FLDict pointer, not a slice. You'll need to call FLValue_ToJSON on it to create a JSON string. That FLSliceResult needs to be released afterwards.

**Review by snej — CHANGES_REQUESTED** [review]
I don't see any changes to the simplistic Python test ... did you run it?

File: CouchbaseLite/Document.py (hunk 1 above)
Comment (bhtaulbee) [reply CouchbaseLite/Document.py]: I took a different approach and simply calling encodeJson instead. Also, I did run the python tests and they ran without error.

**Comment by bhtaulbee** [comment]
I have pushed an update and added comments to the PR. Please review.
Brad
    On Monday, March 14, 2022, 06:23:46 PM MDT, Jens Alfke ***@***.***> wrote:  
 
 


@snej requested changes on this pull request.

I don't see any changes to the simplistic Python test ... did you run it?

In CouchbaseLite/Document.py:
> @@ -78,7 +78,9 @@ def getProperties(self):
 
     @property
     def JSON(self):
-        return sliceToString(lib.CBLDocument_PropertiesAsJSON(self._ref))
+        # TODO: c-api call doesn't exist any more, CBLDocument_PropertiesAsJSON
+        #       need to test the following to see if it works
+        return sliceToString(lib.CBLDocument_Properties(self._ref))

That will not work. CBLDocument_Properties returns a FLDict pointer, not a slice. You'll need to call FLValue_ToJSON on it to create a JSON string. That FLSliceResult needs to be released afterwards.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by snej** [comment]
Thanks!

**Pull request merged**